test/plots/rectanglePlotTests.ts

```diff
@@ -2,444 +2,418 @@
       });
-    };
-
-    it("renders correctly", () => {
-      let xScale = new Plottable.Scales.Linear();
-      let yScale = new Plottable.Scales.Linear();
-      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
-      let rectanglePlot = new Plottable.Plots.Rectangle();
-      rectanglePlot.addDataset(new Plottable.Dataset(DATA));
-      rectanglePlot.x((d) => d.x, xScale)
-                   .y((d) => d.y, yScale)
-                   .x2((d) => d.x2)
-                   .y2((d) => d.y2)
-                   .renderTo(svg);
-      VERIFY_CELLS((<any> rectanglePlot)._renderArea.selectAll("rect"));
-      svg.remove();
-    });
 
-    it("retrieves the correct entity under a point", () => {
-      let xScale = new Plottable.Scales.Linear();
-      let yScale = new Plottable.Scales.Linear();
-      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
-      let dataset = new Plottable.Dataset(DATA);
-      let plot = new Plottable.Plots.Rectangle()
-        .x((d) => d.x, xScale).x2((d) => d.x2)
-        .y((d) => d.y, yScale).y2((d) => d.y2);
-      plot.addDataset(dataset).renderTo(svg);
-      let entities = plot.entitiesAt({ x: xScale.scale(2.5), y: yScale.scale(2.5) });
-      assert.lengthOf(entities, 1, "found only one entity when querying a point inside the third rectangle");
-      assert.strictEqual(entities[0].index, 2, "entity retrieved is at index 2");
-      svg.remove();
-    });
-
-    it("retrieves correct entities under a point", () => {
-      let xScale = new Plottable.Scales.Linear();
-      let yScale = new Plottable.Scales.Linear();
-      let svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
-      let dataset = new Plottable.Dataset([
-        { x: 1, y: 1, x2: 3, y2: 3 },
-        { x: 4, y: 2, x2: 2, y2: 4 }
-      ]);
-      let plot = new Plottable.Plots.Rectangle()
-        .x((d) => d.x, xScale).x2((d) => d.x2)
-        .y((d) => d.y, yScale).y2((d) => d.y2);
-      plot.addDataset(dataset).renderTo(svg);
-      let entities = plot.entitiesAt({ x: xScale.scale(2), y: xScale.scale(2) });
-      assert.lengthOf(entities, 2, "two entities when querying a point in both");
-      assert.strictEqual(entities[0].index, 0, "entity retrieved is at index 0");
-      assert.strictEqual(entities[1].index, 1, "entity retrieved is at index 1");
-      entities = plot.entitiesAt({ x: xScale.scale(4), y: yScale.scale(4) });
-      assert.lengthOf(entities, 1, "found only one entity when querying a point inside the second rectangle");
-      assert.strictEqual(entities[0].index, 1, "entity retrieved is at index 1");
-      entities = plot.entitiesAt({ x: xScale.scale(1), y: yScale.scale(1) });
-      assert.lengthOf(entities, 1, "found only one entity when querying a point inside the first rectangle");
-      assert.strictEqual(entities[0].index, 0, "entity retrieved is at index 0");
-      svg.remove();
-    });
+      afterEach(() => {
+        svg.remove();
+      });
```

Comment:
We made a conscious decision not to put the `svg.remove()` statement in an `afterEach` block since `afterEach` will run even if the test has failed.  This allows us to quickly up the test on the browser to see what has failed visually without modifying the underlying test.

If there is a way to not run `afterEach()` if the test has failed then that works too.